maps/HDRUK/2.2.1/HDRUK/3.0.0/translation.jsonata

```diff
@@ -0,0 +1,293 @@
+    "RESEARCH USE ONLY",
+    "NO LINKAGE"
+];
+
+$allowedDataUseRequirements := [
+    "COLLABORATION REQUIRED",
+    "PROJECT SPECIFIC RESTRICTIONS",
+    "ETHICS APPROVAL REQUIRED",
+    "INSTITUTION SPECIFIC RESTRICTIONS",
+    "GEOGRAPHICAL RESTRICTIONS",
+    "PUBLICATION MORATORIUM",
+    "PUBLICATION REQUIRED",
+    "RETURN TO DATABASE OR RESOURCE",
+    "TIME LIMIT ON USE",
+    "DISCLOSURE CONTROL",
+    "NOT FOR PROFIT USE",
+    "USER SPECIFIC RESTRICTION"
+];
+
+$allowedPurpose := [
+    "STUDY",
+    "DISEASE REGISTRY",
+    "TRIAL",
+    "CARE",
+    "AUDIT",
+    "ADMINISTRATIVE",
+    "FINANCIAL",
+    "STATUTORY",
+    "OTHER"
+];
+
+$allowedSource := [
+    "EPR",
+    "ELECTRONIC SURVEY",
+    "LIMS",
+    "OTHER INFORMATION SYSTEM",
+    "PAPER BASED",
+    "FREETEXT NLP",
+    "MACHINE GENERATED",
+    "OTHER"
+];
+
+$allowedSetting := [
+    "CLINIC",
+    "PRIMARY CARE",
+    "ACCIDENT AND EMERGENCY",
+    "OUTPATIENTS",
+    "IN-PATIENTS",
+    "SERVICES",
+    "COMMUNITY",
+    "HOME",
+    "PRIVATE",
+    "PHARMACY",
+    "SOCIAL CARE",
+    "LOCAL AUTHORITY",
+    "NATIONAL GOVERNMENT",
+    "OTHER"
+];
+
+$allowedPublishingFrequency := [
+    "STATIC",
+    "IRREGULAR",
+    "CONTINUOUS",
+    "BIENNIAL",
+    "ANNUAL",
+    "BIANNUAL",
+    "QUARTERLY",
+    "BIMONTHLY",
+    "MONTHLY",
+    "BIWEEKLY",
+    "WEEKLY",
+    "SEMIWEEKLY",
+    "DAILY",
+    "OTHER"
+];
+
+$allowedTimeLag := [
+    "LESS 1 WEEK",
+    "1-2 WEEKS",
+    "2-4 WEEKS",
+    "1-2 MONTHS",
+    "2-6 MONTHS",
+    "MORE 6 MONTHS",
+    "VARIABLE",
+    "NO TIMELAG",
+    "NOT APPLICABLE",
+    "OTHER"
+];
+
+$allowedDeliveryLeadTime := [
+    "LESS 1 WEEK",
+    "1-2 WEEKS",
+    "2-4 WEEKS",
+    "1-2 MONTHS",
+    "2-6 MONTHS",
+    "MORE 6 MONTHS",
+    "VARIABLE",
+    "NOT APPLICABLE",
+    "OTHER"
+];
+
+$allowedEncodings := [
+    "LOCAL",
+    "OPCS4",
+    "READ",
+    "SNOMED CT",
+    "SNOMED RT",
+    "DM PLUS D",
+    "DM+D",
+    "NHS NATIONAL CODES",
+    "NHS SCOTLAND NATIONAL CODES",
+    "NHS WALES NATIONAL CODES",
+    "ODS",
+    "LOINC",
+    "ICD10",
+    "ICD10CM",
+    "ICD10PCS",
+    "ICD9CM",
+    "ICD9",
+    "ICDO3",
+    "AMT",
+    "APC",
+    "ATC",
+    "CIEL",
+    "HPO",
+    "CPT4",
+    "DPD",
+    "DRG",
+    "HEMONC",
+    "JMDC",
+    "KCD7",
+    "MULTUM",
+    "NAACCR",
+    "NDC",
+    "NDFRT",
+    "OXMIS",
+    "RXNORM",
+    "RXNORM EXTENSION",
+    "SPL",
+    "OTHER"
+];
+
+$tryMatchEnum := function($x, $allowed){(
+    $cleaned := $replace($x, ";", "");
+    $uppered := $uppercase($x);
+    $match := $filter($allowed, function($v, $i, $a){(
+        $allowedUppered := $uppercase($v);
+        $allowedUppered = $uppered ? $a[$i] : null
+    )})
+)};
+
+$cleanEnum := function($x, $allowed){
+    $type($x) = "string" ? [$map($split($x, ","), function($v){
+        $tryMatchEnum($v, $allowed)
+    })] : 
+        [$map($x, function($v){
+            $tryMatchEnum($v, $allowed)
+    })]
+};
```

Comment:
i know youve copy and pasted what i did in reverse. Just not great is it!  Especially when some of the mappings are not just simply switching between lowercase and uppercase...

Chris is aware and hopefully we'll do some cleaning and implement some mappings instead!

Reply:
Agreed, there will be some data loss in the case where we go between HDRv3 and earlier schema versions. It would be good to sort the upgrade to HDRv3 through cleaning and a proper mapping.